Parse/src/main/java/com/parse/Parse.java

```diff
@@ -35,6 +36,86 @@
  * library.
  */
 public class Parse {
+  public static final class Configuration {
+    private final Context context;
+    private final String applicationId;
+    private final String clientKey;
+    private final boolean localDataStoreEnabled;
+    private final List<ParseNetworkInterceptor> interceptors;
+
+    private Configuration(Builder builder) {
+      this.context = builder.context;
+      this.applicationId = builder.applicationId;
+      this.clientKey = builder.clientKey;
+      this.localDataStoreEnabled = builder.localDataStoreEnabled;
+      this.interceptors = builder.interceptors;
+    }
+
+    public static final class Builder {
+      private Context context;
+      private String applicationId;
+      private String clientKey;
+      private boolean localDataStoreEnabled;
+      private List<ParseNetworkInterceptor> interceptors = new ArrayList<>();
+
+      public Builder(Context context) {
+        this.context = context;
+      }
+
+      public Builder applicationId(String applicationId) {
+        this.applicationId = applicationId;
+        return this;
+      }
+
+      public Builder clientKey(String clientKey) {
+        this.clientKey = clientKey;
+        return this;
+      }
+
+      public Builder addNetworkInterceptor(ParseNetworkInterceptor interceptor) {
+        interceptors.add(interceptor);
+        return this;
+      }
+
+      public Builder enableLocalDataStore() {
+        localDataStoreEnabled = true;
+        return this;
+      }
+
+      /* package */ Builder setNetworkInterceptors(Collection<ParseNetworkInterceptor> interceptors) {
+        interceptors.clear();
+        if (interceptors != null) {
+          interceptors.addAll(interceptors);
+        }
+        return this;
+      }
+
+      public Configuration build() {
+        return new Configuration(this);
+      }
+    }
+
+    public Context context() {
+      return context;
+    }
+
+    public String applicationId() {
+      return applicationId;
+    }
+
+    public String clientKey() {
+      return clientKey;
+    }
+
+    public boolean isLocalDataStoreEnabled() {
+      return localDataStoreEnabled;
+    }
+
+    public List<ParseNetworkInterceptor> networkInterceptors() {
+      return interceptors;
+    }
```

Comment:
I don't see accessors being useful and we can always add them back if there's a demand